.github/workflows/ci.yml

```diff
@@ -177,14 +177,35 @@ jobs:
           HATCH_BUILD_CLEAN: "true"
         run: python -m build
 
-      - name: Save release artifacts (for inspection)
+      - name: Store the distribution packages
         uses: actions/upload-artifact@v4
         with:
-          name: dist
+          name: python-package-distributions
           path: dist/
 
-      - name: Publish release
-        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+  ############################################################################
+  # (Possibly) Publish to PyPI
+  ############################################################################
+  publish-to-pypi:
+    name: >-
+      Publish Python 🐍 distribution 📦 to PyPI
+    if: github.ref_type == 'tag'  # only publish to PyPI on tag pushes
+    needs:
+      - build
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/Lektor
+    permissions:
+      id-token: write  # IMPORTANT: mandatory for trusted publishing
+    steps:
+      - name: Download all the dists
+        uses: actions/download-artifact@v4
+        with:
+          name: python-package-distributions
+          path: dist/
+
+      - name: Publish distribution 📦 to PyPI
         uses: pypa/gh-action-pypi-publish@release/v1
         with:
-          password: ${{ secrets.PYPI_PASSWORD }}
+          password: ${{ secrets.PYPI_PASSWORD }}
```

Comment:
Urgh.. I wonder why my editor dropped the trailing LF here. It doesn't usually break text files like that.